src/components/dialog/Dialog.jsx

```diff
@@ -196,6 +208,7 @@ function BaseDialog({
       onClick={onDismiss ? handleOverlayClick : undefined}
       onKeyUp={handleKeyUp}
       ref={overlayRef}
+      id={id}
```

Comment:
why do we need this?

Reply:
oh I see now, because of stop propagation we weren't able to catch event outside dialog.